### PR TITLE
Update auto approve dependency PR to run on workflow completion

### PR DESCRIPTION
**Pull request by gsheni**
- This will make it so that when a workflow (unit tests) completes on a branch, the auto approve workflow will run.

### CODE REVIEW

File: .github/workflows/auto_approve_dependency_PRs.yml [hunk 1]
```diff
@@ -3,6 +3,13 @@ on:
   schedule:
     - cron: '*/5 * * * *'
   workflow_dispatch:
+  workflow_run:
```
Comment (gsheni) [inline .github/workflows/auto_approve_dependency_PRs.yml:6]: https://docs.github.com/en/actions/learn-github-actions/events-that-trigger-workflows#workflow_run

File: .github/workflows/auto_approve_dependency_PRs.yml [hunk 2]
```diff
@@ -3,6 +3,13 @@ on:
...
+    workflows: ["Unit Tests - Latest Dependencies"]
+    branches:    
+      - 'latest-dep-update-[a-f0-9]+'
```
Comment (gsheni) [inline .github/workflows/auto_approve_dependency_PRs.yml:9]: Need a double check on this syntax
- https://docs.github.com/en/actions/learn-github-actions/workflow-syntax-for-github-actions#filter-pattern-cheat-sheet
```
dep_update = '^latest-dep-update-[a-f0-9]{7}$'
min_dep_update = '^min-dep-update-[a-f0-9]{7}$'
```

File: .github/workflows/auto_approve_dependency_PRs.yml (hunk 1 above)
Comment (rwedge) [reply .github/workflows/auto_approve_dependency_PRs.yml:6]: Can we remove the "schedule" event and rely on workflow_run only?

File: .github/workflows/auto_approve_dependency_PRs.yml (hunk 2 above)
Comment (rwedge) [reply .github/workflows/auto_approve_dependency_PRs.yml:7]: Not sure if the github syntax supports `[a-f0-9]{7}` but we could put `[a-f0-9]` 7 times in a row . . .

Comment (gsheni) [reply .github/workflows/auto_approve_dependency_PRs.yml:7]: That's why I put `+`, which means `+: Matches one or more of the preceding character.`.

File: .github/workflows/auto_approve_dependency_PRs.yml (hunk 1 above)
Comment (gsheni) [reply .github/workflows/auto_approve_dependency_PRs.yml:4]: Removed

**Review by rwedge — APPROVED** [review]
LGTM

**Pull request merged**